source/credentials_provider_ecs.c

```diff
@@ -545,11 +545,10 @@ static void s_credentials_provider_ecs_destroy(struct aws_credentials_provider *
         return;
     }
 
-    impl->function_table->aws_http_connection_manager_release(impl->connection_manager);
-
     aws_string_destroy(impl->path_and_query);
     aws_string_destroy(impl->auth_token);
     aws_tls_connection_options_clean_up(&impl->connection_options);
+    impl->function_table->aws_http_connection_manager_release(impl->connection_manager);
```

Comment:
Release will call on_connection_manager_shutdown, which will do memory free for provider and impl. We should free impl related first, then call release. Same to x509, sts_web_identity.

Reply:
Please add this as a comment here. This is important to the next reader.